### PR TITLE
ISPN-12989 Initial state transfer for cross-site replication

### PR DESCRIPTION
**Pull request by pruivo**
Listens to CacheStarted events and if the cache has an async backup to SiteB, then it requests state.

https://issues.redhat.com/browse/ISPN-12989

### CODE REVIEW

**Comment by ryanemerson** [comment]
@pruivo Do we need to update the docs to reflect the new behaviour?

**Comment by pruivo** [comment]
sure @ryanemerson

File: documentation/src/main/asciidoc/topics/con_xsite_state_transfer.adoc [hunk 1]
```diff
@@ -53,6 +53,10 @@ To avoid temporary network disconnects triggering state transfer operations, the
 The status of a backup location must be offline and it must not be included in the cross-site view with JGroups RELAY2.
 ====
 
+The automatic state transfer is not only triggered when the network is broken, but also when a cache starts.
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/con_xsite_state_transfer.adoc:56]: @infinispan/docs can you take a look, please?

Comment (domiborges) [reply documentation/src/main/asciidoc/topics/con_xsite_state_transfer.adoc]: few minor edits otherwise lgtm!

**Comment by ryanemerson** [comment]
Thanks @pruivo

**Pull request merged**